Firmware/first_lay_cal.cpp

```diff
@@ -37,12 +37,30 @@ static constexpr float spacing(float layer_height, float extrusion_width, float
 static void lay1cal_common_enqueue_loop(const char * const * cmd_sequence, const uint8_t steps) {
     for (uint8_t i = 0; i < steps; ++i)
     {
-        enquecommand_P(static_cast<char*>(pgm_read_ptr(cmd_sequence + i)));
+        void * pgm_ptr = pgm_read_ptr(cmd_sequence + i);
+
+        // M702 is currently only used with MMU enabled
+        if (pgm_ptr == MSG_M702 && !MMU2::mmu2.Enabled()) {
+            continue;
+        }
+
+        enquecommand_P(static_cast<char*>(pgm_ptr));
     }
 }
 
-static const char extrude_fmt[] PROGMEM = "G1 X%d Y%d E%-.5f";
-static const char zero_extrusion[] PROGMEM = "G92 E0";
+static const char extrude_fmt_X[] PROGMEM = "G1X%gE%g";
+static const char extrude_fmt_Y[] PROGMEM = "G1Y%gE%g";
+static const char zero_extrusion[] PROGMEM = "G92E0";
+static const char feedrate_F1080[] PROGMEM = "G1F1080";
+#ifndef NEW_FIRST_LAYER_CAL
+int8_t invert = 1;
+const float short_length = 20;
+#else
+int8_t invert = -1;
```

Comment:
Is there any point having the same variable with the same value here regardless if `NEW_FIRST_LAYER_CAL` is defined or not.

Can `int8_t invert = 1;` be moved out of the `ifdef`? It doesn't save any memory, but its shorter code.

Reply:
The values are `-1`  vs `1` to invert the direction